libtsuba/src/RDGCore.cpp

```diff
@@ -335,7 +335,7 @@ katana::RDGCore::RemoveEdgeProperty(int i, katana::TxnContext* txn_ctx) {
 
 katana::Result<katana::NUMAArray<katana::EntityTypeID>>
 katana::RDGCore::node_entity_type_id_array(size_t begin, size_t end) const {
-  if (part_header().IsEntityTypeIDsOutsideProperties()) {
+  if (!part_header().IsUint16tEntityTypeIDs()) {
```

Comment:
This seems wrong to me. Why are you returning an empty array when the types are 8 bits? You are also trying to load the array when there is no independent type array and therefore no file.

Reply:
since storage_format_version_2 support was dropped a while ago, uint8_t EntityTypeIDs are unsupported, so there is no need to check for them. 
The naming for these `IsBlah` methods is kinda bad imo, and that can only be blamed on me. I'll have to revisit them later. 

to clarify: 
`IsEntityTypeIDsOutsideProperties` basically just says we have a storage_format_version 2 or higher rdg
`IsUint16tEntityTypeIDs` basically just says we have a storage_format_version 3 or higher rdg, so it is a "stronger" check than `IsEntityTypeIDsOutsideProperties`

Because of all of that, only RDGs >= storage_format_version_3 will have EntityTypeIDs outside properties, so this logic checks out.

Reply:
Can we rename the functions then? Because `IsUint16tEntityTypeIDs()` is confusing to read for me. Also if we don't support 8 bit ids any more then `LoadAndMaybeConvert` should be renamed to `LoadIDArray` and it doesn't need to be templated.

Reply:
yep, we definitely can. I'm not going to in this PR though, its gone on long enough :) 

I did rename and remove the templating from `LoadAndMaybeConvert()` though

Reply:
Nice, thanks